lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -31,8 +31,44 @@ def colorized_message_lines(colorizer=::RSpec::Core::Formatters::ConsoleCodes)
           end
         end
 
-        def formatted_backtrace
-          backtrace_formatter.format_backtrace(exception_backtrace, example.metadata)
+        def formatted_backtrace(exception=@exception)
+          lines = backtrace_formatter.format_backtrace((exception.backtrace || []), example.metadata)
+          potential_lines = formatted_cause(exception)
+
+          lines += potential_lines unless potential_lines.empty?
+
+          lines
+        end
+
+        if RSpec::Support::RubyFeatures.supports_exception_cause?
+          def formatted_cause(exception)
+            exceptions = []
+
+            while exception.respond_to?(:cause) && exception.cause
```

Comment:
Is the `respond_to?` check here necessary?  Under what conditions would `Exception.method_defined?(:cause)` return true (that's the logic for the `supports_exception_cause?` check) and `exception.respond_to?(:cause)` return false?